apps/prairielearn/src/sync/fromDisk/assessments.js

```diff
@@ -315,6 +332,80 @@ module.exports.sync = async function (courseId, courseInstanceId, assessments, q
       }
     });
   }
+
+  /** @type {Set<string>} */
+  const importedQids = new Set();
+  /** @type {Map<string, string[]>} */
+  const importedQidAssessmentMap = new Map();
+  Object.entries(assessments).forEach(([tid, assessment]) => {
+    if (!assessment.data) return;
+    (assessment.data.zones || []).forEach((zone) => {
+      (zone.questions || []).forEach((question) => {
+        let qids = question.alternatives
+          ? question.alternatives.map((alternative) => alternative.id)
+          : [];
+        if (question.id) {
+          qids.push(question.id);
+        }
+        qids.forEach((qid) => {
+          if (qid[0] === '@') {
+            importedQids.add(qid);
+            let tids = importedQidAssessmentMap.get(qid);
+            if (!tids) {
+              tids = [];
+              importedQidAssessmentMap.set(qid, tids);
+            }
+            tids.push(tid);
+          }
+        });
+      });
+    });
+  });
+
+  if (importedQids.size > 0) {
+    let institutionId = await sqldb.queryRow(
+      sql.get_institution_id,
+      { course_id: courseId },
+      z.string(),
+    );
+    let questionSharingEnabled = await features.enabled('question-sharing', {
+      course_id: courseId,
+      course_instance_id: courseInstanceId,
+      institution_id: institutionId,
+    });
+    if (!questionSharingEnabled && config.checkSharingOnSync) {
```

Comment:
Nit: I'd be inclined to put essentially all the new code here inside an `if (config.checkSharingOnSync) { ... }` conditional just so it's clear that all of this only matters if that config is set. This would match what we do with `config.checkAccessRulesExamUuid` above.

Reply:
That totally makes sense for better readability/organization. Done.

Reply:
Oh wait, we can't but all of this behind a flag because syncing the `assessmentConfig.json`s will fail if we don't add the shared questions to the questions data structure like we do here. Let me see if I can figure something better out.

Reply:
Ya, basically everything here except for actually reporting the errors needs to happen so that the `questionIds` dict has the proper info when it gets passed to `getParamsForAssessment` later on in this function. I'm just going to leave it how it was.